### ISSUE TITLE
Build passes with testthat errors

### ISSUE THREAD

**Issue by pat-s** [r-lib/rcmdcheck#69]
See https://travis-ci.org/mlr-org/mlr/jobs/392363952

We used the dev version of `rcmdcheck`.

cross-ref https://github.com/ropenscilabs/tic/issues/44, https://github.com/mlr-org/mlr/pull/2295

**Comment by gaborcsardi** [comment]
this https://travis-ci.org/mlr-org/mlr/jobs/392363952#L2090:  
```
── Test failures ───────────────────────────────────────────── run-classif1 ────
```
I believe, so this is strange. CAn you download the output of `R CMD check` somehow?

**Comment by gaborcsardi** [comment]
Oh, yeah, that's  not  printed by rcmdcheck, of course. Anyway, is it possible to download the raw output?

**Comment by pat-s** [comment]
> Anyway, is it possible to download the raw output?

You mean running `R CMD check` locally and pasting it here?

**Comment by gaborcsardi** [comment]
Well, I don't really want to run it locally for mlr, takes too long, and the failures might not be deterministic, anyway. I mean that after 
```
x <- rcmdcheck()
```
the original standard output and error are in  `x$stdout` and `x$stderr`. Can I get these from Travis. Ideally tic could save these as artifacts. Or save `x` itself.

Essentially I am asking for a smaller reproducible example.

**Comment by pat-s** [comment]
Ok, we can try to do that. I can upload the artifacts to AWS but how do I get out `x` in an elegant way?
Saving it after the call in an `.rda` file and then uploading this does not seem very intuitive. 

@krlmlr Can you help here?

**Comment by pat-s** [comment]
Running this problematic [build](https://travis-ci.org/mlr-org/mlr/jobs/396403783) locally works fine. So the problem seems to be travis specific? 
Trying to get out `x <- rcmdcheck()` as an artifact now..

```r
     OK: 3588 SKIPPED: 0 FAILED: 1
     1. Error: tuning allows usage of budget (@test_base_tuning.R#120) 
     
     Error: testthat unit tests failed
     Execution halted
✔  checking for unstated dependencies in vignettes
✔  checking package vignettes in ‘inst/doc’
─  checking running R code from vignettes ...
      ‘mlr.Rmd’ using ‘UTF-8’ ... OK
    NONE
✔  checking re-building of vignette outputs
✔  checking PDF version of manual

   
── 1 error ✖ | 1 warning ✖ | 0 notes ✔
```

**Comment by pat-s** [comment]
@gaborcsardi I finally have an object to debug: https://www.dropbox.com/s/bvmiswmzvetm1he/rcmdcheck.rda?dl=0

It's from [this](https://travis-ci.org/mlr-org/mlr/jobs/400104073) build.

**Comment by pat-s** [comment]
@gaborcsardi This little change helps catching the error in the summary. Subsequently, `tic` will also correctly end the build with `1`.

https://github.com/pat-s/rcmdcheck/blob/10c6e09e5d86ebb60c13358419aa60de7d9d0b5b/R/parse.R#L30

Not sure if this is completely safe for all other "ERROR" appearances but it works for errors in tests.

Catching the errors in tests is crucial, would be great if we could get this to CRAN soon. 
Even though its within #72, a hotfix integrating this would be highly appreciated.

**Comment by gaborcsardi** [comment]
This was fixed in c3edf23d77a771751b1d30a39af2a8d774c864e2
